test/Core/Eval/Base.hs

```diff
@@ -95,6 +97,18 @@ coreEvalAssertion' mode tab mainFile expectedFile step =
               )
       Nothing -> assertFailure ("No main function registered in: " <> toFilePath mainFile)
   where
+    sym = fromJust (tab ^. infoMain)
+
+    mkArg :: Type -> Text -> Node
+    mkArg ty arg =
+      let n = fst $ fromRight' $ decimal arg
+       in if
+              | isTypeBool ty ->
+                  if
+                      | n == 0 -> mkConstr' (BuiltinTag TagFalse) []
+                      | otherwise -> mkConstr' (BuiltinTag TagTrue) []
+              | otherwise -> mkConstant' (ConstInteger n)
```

Comment:
```suggestion
       in case (isTypeBool ty, n) of
            (True, 0) -> mkConstr' (BuiltinTag TagFalse) []
            (True, _) -> mkConstr' (BuiltinTag TagTrue) []
            (False, _) -> mkConstant' (ConstInteger n)
```